src/test/java/ca/uhn/fhir/jpa/starter/interceptor/CustomSearchNarrowingInterceptorTest.java

```diff
@@ -0,0 +1,205 @@
+package ca.uhn.fhir.jpa.starter.interceptor;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mockStatic;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.hl7.fhir.instance.model.api.IBaseResource;
+import org.hl7.fhir.r4.model.Observation;
+import org.hl7.fhir.r4.model.Patient;
+import org.hl7.fhir.r4.model.Resource;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.mockito.InjectMocks;
+import org.mockito.MockedStatic;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ca.uhn.fhir.context.FhirContext;
+import ca.uhn.fhir.jpa.starter.AppProperties;
+import ca.uhn.fhir.jpa.starter.AppProperties.Oauth;
+import ca.uhn.fhir.jpa.starter.util.OAuth2Helper;
+import ca.uhn.fhir.model.api.IQueryParameterOr;
+import ca.uhn.fhir.model.api.IQueryParameterType;
+import ca.uhn.fhir.rest.annotation.OptionalParam;
+import ca.uhn.fhir.rest.annotation.Search;
+import ca.uhn.fhir.rest.api.server.RequestDetails;
+import ca.uhn.fhir.rest.client.api.IGenericClient;
+import ca.uhn.fhir.rest.param.BaseAndListParam;
+import ca.uhn.fhir.rest.param.ReferenceAndListParam;
+import ca.uhn.fhir.rest.param.TokenAndListParam;
+import ca.uhn.fhir.rest.server.IResourceProvider;
+import ca.uhn.fhir.rest.server.interceptor.auth.AuthorizedList;
+import ca.uhn.fhir.test.utilities.server.RestfulServerExtension;
+
+class CustomSearchNarrowingInterceptorTest {
+
+	@Spy
+	AppProperties mockConfig;
+
+	@Spy
+	Oauth mockOAuth;
+
+	@InjectMocks
+	CustomSearchNarrowingInterceptor ourSearchNarrowingInterceptor;
+
+	private static Logger logger = LoggerFactory.getLogger(CustomSearchNarrowingInterceptorTest.class);
+	private static List<Resource> ourReturn;
+	private static String ourLastHitMethod;
+	private static TokenAndListParam ourLastIdParam;
+	private static ReferenceAndListParam ourLastPatientParam;
+	private static final FhirContext ourCtx = FhirContext.forR4();
+	private static final String PATIENT_ID = "12345";
+	public IGenericClient myClient;
+
+	@RegisterExtension
+	public RestfulServerExtension myRestfulServerExtension = new RestfulServerExtension(ourCtx)
+			.registerProvider(new MockPatientResourceProvider())
+			.registerProvider(new MockObservationResourceProvider());
+
+	@BeforeEach
+	public void before() throws IOException {
+		MockitoAnnotations.openMocks(this);
+		ourReturn = Collections.emptyList();
+		ourLastHitMethod = null;
```

Comment:
Why would you want to test the "last hit method"?

Reply:
I have followed the [upstream repo tests](https://github.com/hapifhir/hapi-fhir/blob/fb3e141c4d7b790b824132b833c4bddc52e03835/hapi-fhir-structures-r4/src/test/java/ca/uhn/fhir/rest/server/interceptor/auth/SearchNarrowingInterceptorTest.java#L65), as when we search for fhirResource( observation or patient ), the variable `ourLastHitMethod` will return the method name we have hit so it helps to verify right method has been called during execution.